### PR TITLE
Add script for getting all authors contributing to organization.

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
## Main changes of this PR

Adds a script to parse all repos of an orgnization for commit authors.

Must not end up in this repository and not necessarily in `tools`. @MakisH you probably have a better overview.

## Motivation and additional information

Helps us to create a list with names of all contributors.

## Reviewers' checklist

* [ ] Do you understand the code changes?

### CODE REVIEW

**Comment by MakisH** [comment]
Thank you, this is useful! Can you also add a `README.md` on what it does and how to use it?

**Comment by BenjaminRodenberg** [comment]
> Thank you, this is useful! Can you also add a `README.md` on what it does and how to use it?

I added documentation for the `--help` flag. Providing an individual `README.md` means additional maintainment and is not really needed just for a script in my opinion. Therefore, I would like to avoid it.

File: tools/community/list_collabs.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            # Traverse all commits on page
+            for commit in response.json():
+                number_commits += 1
+                try:
+                    # extract name of author
+                    author = commit['author']['login']
+                    commit_authors.add(author)
+                except TypeError:
+                    # need to skip many commits. Not sure why.
+                    skipped_commits.append(commit)
```
Comment (MakisH) [inline tools/community/list_collabs.py:73]: Would be good to store these commits in a file or show them with some option. Running this I got 1949 skipped commits, a number that sounds a bit alarming and generates questions.

File: tools/community/list_collabs.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        # While page has entries, continue
+        while len(response.json()) > 0:
+            print("On page {} got {} hits.".format(pageno, len(response.json())))
+            pageno += 1
+            api_url = "https://api.github.com/repos/{}/{}/commits?per_page=100&page={}".format(organization, repo, pageno)
+            if '--user' in arguments.keys():
+                response = requests.get(api_url, auth=HTTPBasicAuth(arguments['--user'], arguments['--token']))
+            else:
+                response = requests.get(api_url)
```
Comment (MakisH) [inline tools/community/list_collabs.py:63]: We should also check the response for being valid. When running without a user/password, I got stuck in a "On page (increasing) got 2 hits.", but no error. I had to guess that this is related to the access.

**Comment by MakisH** [comment]
> > Thank you, this is useful! Can you also add a `README.md` on what it does and how to use it?
> 
> I added documentation for the `--help` flag. Providing an individual `README.md` means additional maintainment and is not really needed just for a script in my opinion. Therefore, I would like to avoid it.

But every other folder in `tools/` has a `README.md`, explaning what these scripts are.
I am not sure this is the right place, but I can also not think of any better place at the moment.

**Comment by MakisH** [comment]
@BenjaminRodenberg I still believe that this is not really the right place for this script. Important is that it stays documented somewhere. My original plan was to make a different script that can give us a ratio of external/total contributions. But this had merely academic intention.

**Pull request closed without merging**